test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ODataDynamicPropertyTests.cs

```diff
@@ -0,0 +1,785 @@
+            // "Name" dynamic property ignored
+            var expectedResult = "{\"@odata.type\":\"#ServiceNS.Director\",\"Id\":1,\"Name\":\"Director 1\"}";
+
+            VerifyMessageBody(expectedResult, messageBody);
+        }
+
+        [TestMethod]
+        public void SerializationIgnoresEntityTypeAsDynamicProperty()
+        {
+            var director = new Director { Id = 1, Name = "Director 1" };
+            director.DynamicProperties.Add("Producer", new Producer { Id = 2, Name = "Producer 2" });
+
+            var messageBody = SerializeEntity(this.directorEntitySetName, director);
+            // "Producer" entity type dynamic property ignored
+            var expectedResult = "{\"@odata.type\":\"#ServiceNS.Director\",\"Id\":1,\"Name\":\"Director 1\"}";
+
+            VerifyMessageBody(expectedResult, messageBody);
+        }
+
+        [TestMethod]
+        public void SerializationIgnoresDynamicPropertiesDictionaryWithNoContainerPropertyAttribute()
+        {
+            var editor = new Editor { Id = 1, Name = "Editor 1" };
+            editor.DynamicProperties.Add("Title", "Dr");
+
+            var messageBody = SerializeEntity(this.editorEntitySetName, editor);
+            // Nothing in the dictionary property is serialized
+            var expectedResult = "{\"@odata.type\":\"#ServiceNS.Editor\",\"Id\":1,\"Name\":\"Editor 1\"}";
+
+            VerifyMessageBody(expectedResult, messageBody);
+        }
+
+        [TestMethod]
+        public void SerializationIgnoresDynamicPropertiesDictionaryIfEdmTypeIsNotOpen()
+        {
+            // Producer is not an open type but client type has a dictionary of string and object
+            var producer = new Producer { Id = 1, Name = "Producer 1" };
+            producer.DynamicProperties.Add("Title", "Ms");
+
+            var messageBody = SerializeEntity(this.producerEntitySetName, producer);
+            // Nothing in the dictionary property is serialized
+            var expectedResult = "{\"@odata.type\":\"#ServiceNS.Producer\",\"Id\":1,\"Name\":\"Producer 1\"}";
+
+            VerifyMessageBody(expectedResult, messageBody);
+        }
+
+        [TestMethod]
+        public void MaterializationWithNoDynamicProperty()
+        {
+            var rawJsonResponse = "{" +
+                "\"@odata.context\":\"http://tempuri.org/$metadata#Directors/$entity\"," +
+                "\"Id\":1,\"Name\":\"Director 1\"" +
+                "}";
+
+            var materializedObject = MaterializeEntity<Director>(rawJsonResponse);
+
+            Assert.IsNotNull(materializedObject);
+            Assert.AreEqual(materializedObject.Id, 1);
+            Assert.AreEqual(materializedObject.Name, "Director 1");
+            Assert.AreEqual(materializedObject.DynamicProperties.Count, 0);
+        }
+
+        private void AssertShared(Director materializedObject, Type dynamicPropertyType, string dynamicPropertyName)

```

Comment:
Why the name `AssertShared`?

Reply:
@habbes It happens that the first set of asserts in each test scenario were exactly the same and I thought that instead of repeating them over and over for each test I'd put them in a shared method that each test calls.

Reply:
Ooh. I thought it was "shared" in the context of static members or shared data. Maybe "AssertCommon"? But it's non-issue

Reply:
@habbes I struggled with the naming. I could name it `AssertCommon` as I'm adding the test you requested